src/pages/[id]/index.js

```diff
@@ -19,23 +19,35 @@ import {
   HEIGHT_40PX,
 } from '@constants/profile-image';
 import { useEffect, useState } from 'react';
+import { userContext } from '@store/user/user-context';
 
 const MemberProfile = ({ imageLink, user, contributions, errorMessage }) => {
+  const { isSuperUserMode } = userContext();
   const [activeTasksData, setActiveTasksData] = useState([]);
   const router = useRouter();
   const { id } = router.query;
-  useEffect(() => {
-    (async () => {
-      const tasksURL = getActiveTasksURL(id);
-      const tasksResponse = await fetch(tasksURL);
-      const { tasks } = await tasksResponse.data;
-      setActiveTasksData(tasks);
-    })();
-  }, [id]);
 
   if (errorMessage) {
     return <NotFound errorMsg={errorMessage} />;
   }
+  if (!user.roles?.member && !isSuperUserMode) {
+    return <NotFound errorMsg="Unauthorized to view this page." />;
+  }
+
+  const fillActiveTasksArray = async () => {
+    const tasksURL = getActiveTasksURL(id);
+    try {
+      const tasksResponse = await fetch(tasksURL);
+      const { tasks } = await tasksResponse.data;
+      setActiveTasksData(tasks);
+    } catch {
+      setActiveTasksData([]);
+    }
+  };
```

Comment:
Could you check more on useCallback?
and we could go ahead and add `fillActiveTasksArray` in the dependency there

Reply:
https://dmitripavlutin.com/dont-overuse-react-usecallback/

I love these blogs along with official docs ofcourse :)

Reply:
Thanks for the blog Harshith. It was indeed helpful. I have added the fillActiveTasksArray in the dependency array & wrapped fillActiveTasksArray in useCallBack.